scene/main/canvas_item.cpp

```diff
@@ -1401,6 +1436,7 @@ void CanvasItem::set_visibility_layer(uint32_t p_visibility_layer) {
 	ERR_THREAD_GUARD;
 	visibility_layer = p_visibility_layer;
 	RenderingServer::get_singleton()->canvas_item_set_visibility_layer(canvas_item, p_visibility_layer);
+	update_configuration_warnings();
```

Comment:
The child items' warnings might be affected/changed as a result. Does this propagate the update for the children? :thinking: